### DISCUSSION TITLE
Why not Flight SQL?

### DISCUSSION THREAD

**Discussion by kPusta19** [CeresDB/ceresdb#1602] [Q&A] [answered]
Are there any plans to add Arrow Flight SQL?

I think HoraeDB usage cases will often be related to performance requirements. In this case, it would be very useful for many architectural solutions to add support for Flight SQL, since HoraeDB relies on Arrow technology to minimize serialization/deserialization of data.

I've seen that the SDK uses its own gRPC services, which is also have great performance, but it seems like Flight RPC (SQL) would be logical to use.

**Comment by jiacai2050** [comment, marked as answer]
Sorry, we don't have plans for FlightSQL.

HoraeDB is mainly used as a prometheus remote storage,  so the main API to insert data is via [remote write](https://prometheus.io/docs/specs/remote_write_spec_2_0/) and query via [remote read](https://prometheus.io/docs/prometheus/latest/querying/remote_read_api/).
